src/fidesops/service/privacy_request/request_service.py

```diff
@@ -0,0 +1,73 @@
+import logging
+from datetime import datetime
+from typing import Optional, Any, Dict, Set, List
+
+from sqlalchemy.orm import Session
+
+from fidesops.models.policy import Policy, ActionType
+from fidesops.models.privacy_request import PrivacyRequestStatus, PrivacyRequest
+from fidesops.schemas.drp_privacy_request import DrpPrivacyRequestCreate
+from fidesops.schemas.masking.masking_configuration import MaskingConfiguration
+from fidesops.schemas.masking.masking_secrets import MaskingSecretCache
+from fidesops.schemas.policy import Rule
+from fidesops.schemas.redis_cache import PrivacyRequestIdentity
+from fidesops.service.masking.strategy.masking_strategy_factory import get_strategy
+
+logger = logging.getLogger(__name__)
+
+
+def retrieve_policy(db: Session, field: str, value: Any) -> Optional[Policy]:
+    return Policy.get_by(
+        db=db,
+        field=field,
+        value=value,
+    )
+
+
+def build_required_privacy_request_kwargs(
+    requested_at: Optional[datetime], policy_id: str
+) -> Dict[str, Any]:
+    return {
+        "requested_at": requested_at,
+        "policy_id": policy_id,
+        "status": PrivacyRequestStatus.pending.value,
+    }
+
+
+def create_privacy_request(db: Session, kwargs: Dict[str, Any]):
+    return PrivacyRequest.create(db=db, data=kwargs)
+
+
+def cache_data(
+    privacy_request: PrivacyRequest,
+    policy: Policy,
+    identity: PrivacyRequestIdentity,
+    encryption_key: Optional[str],
+    drp_request_body: Optional[DrpPrivacyRequestCreate],
+) -> None:
+    # Store identity and encryption key in the cache
+    logger.info(f"Caching identity for privacy request {privacy_request.id}")
+    privacy_request.cache_identity(identity)
```

Comment:
DRP identity will come in as encrypted JWT. We will decrypt before calling this method so that the format is consistent with current functionality